modules/user_accounts/php/my_preferences.class.inc

```diff
@@ -0,0 +1,1134 @@
+                                    $DB->insert(
+                                        'users_notifications_rel',
+                                        array(
+                                         "module_id"  => $module_id,
+                                         "service_id" => $service_id,
+                                         "user_id"    => $user->getData("ID"),
+                                        )
+                                    );
+                                }
+                            } else {
+                                if ($subscribed === 'Y') {
+                                    $DB->delete(
+                                        'users_notifications_rel',
+                                        array(
+                                         "module_id"  => $module_id,
+                                         "service_id" => $service_id,
+                                         "user_id"    => $user->getData("ID"),
+                                        )
+                                    );
+                                }
+                            }
+                            unset($values[$var_name]);
+                        }
+                    }
+                }
+            }
+        }
+        // END NOTIFICATIONS UPDATE
+
+        // make the set
+        foreach ($values as $key => $value) {
+            if (!empty($value)) {
+                $set[$key] = $value;
+            } else {
+                $set[$key] = null;
+            }
+        }
+
+        // update the user
+        if ($this->isCreatingNewUser()) {
+            // insert a new user
+            $success = \User::insert($set);
+            $user    = \User::factory($set['UserID']);
+            $uid     = $user->getData('ID');
+            foreach ($us_curr_sites as $site) {
+                $DB->insert(
+                    'user_psc_rel',
+                    array(
+                     "UserID"   => $uid,
+                     "CenterID" => $site,
+                    )
+                );
+            }
+        } else {
+            // update the user
+            $user    = \User::factory($this->identifier);
+            $success = $user->update($set);
+        }
+
+        // prepend two random characters
+        if (isset($set['Password_hash'])) {
+            // Update CouchDB. Must do before password is salted/hashed.
+            $expiry = isset($values['Password_expiry'])
+                ? $values['Password_expiry'] : null;
+            $user->updatePassword($set['Password_hash'], $expiry);
+        }
+
+        // update the user permissions if applicable
+        // If the user is editing his/her own account, skip the part where
+        // changes to the permissions are handled (there should not be any
+        // change to the user's permission set since all the checkboxes are
+        // disabled)
+        if (!$this->_isEditingOwnAccount()) {
```

Comment:
the permissions part should never be applicable, I think this whole block can be removed.

Reply:
agree